deleting/deleting-documents-in-a-mongodb-collection-with-go.md

```diff
@@ -0,0 +1,163 @@
+
+Let's say that we want to delete a single document from one of our collections. We can make use of the `DeleteOne` function and provide a filter for the document that should be deleted. Take the following for example:
+
+```go
+database := client.Database("quickstart")
+podcastsCollection := database.Collection("podcasts")
+result, err := podcastsCollection.DeleteOne(ctx, bson.M{"title": "The Polyglot Developer Podcast"})
+if err != nil {
+    log.Fatal(err)
+}
+fmt.Printf("DeleteOne removed %v document(s)\n", result.DeletedCount)
+```
+
+The above code obtains a handle to the database and collection that we wish to use from the client connection. Using the `DeleteOne` function, we can provide an application context as well as a filter. The filter in this example is around a podcast with a particular title.
+
+If there was an error, we catch it and terminate the application, otherwise we use the result to print how many documents were deleted. Because we are using the `DeleteOne` function, we can only ever have a `DeletedCount` of zero or one.
+
+Establishing a connection to the cluster and defining an application context can be seen in a [previous tutorial](https://www.mongodb.com/blog/post/quick-start-golang--mongodb--starting-and-setup) of this getting started series.
+
+## Deleting Many Documents from a MongoDB Collection
+
+There will often be scenarios where you need to remove more than one document from a collection in a single operation. For these tasks we can make use of the `DeleteMany` function on a collection, which behaves similar to the `InsertMany` and `UpdateMany` operations that we have already seen.
+
+To see this in action, take a look at the following code:
+
+```go
+database := client.Database("quickstart")
+episodesCollection := database.Collection("episodes")
+result, err = episodesCollection.DeleteMany(ctx, bson.M{"duration": 25})
+if err != nil {
+    log.Fatal(err)
+}
+fmt.Printf("DeleteMany removed %v document(s)\n", result.DeletedCount)
+```
+
+In the above example, pretty much everything is the same. The exception is that we're using a different collection and we're using the `DeleteMany` function. In this example, any document that has a field `duration` that equals `25`, they will be deleted.
+
+In both the `DeleteOne` and `DeleteMany` examples both range and equality filters can be used. More information on the available operators can be found in the [documentation](https://docs.mongodb.com/manual/reference/operator/query/).
+
+## Dropping a MongoDB Collection and All Documents within the Collection
+
+Removing a single document or many documents isn't the only option. Entire collections can be dropped which would remove all documents in the collection without using a filter. An example of this can be seen below:
+
+```go
+if err = podcastsCollection.Drop(ctx); err != nil {
+    log.Fatal(err)
+}
+```
+
+Dropping an entire collection will return an error if something has failed.
```

Comment:
You should mention here that non-existent collections can be dropped. The driver will mask the server error in this case and return a `nil` error.